src/math/deconvolution_2d.jl

```diff
@@ -0,0 +1,1069 @@
+
+Well images in computers work exactly the same way! Each image is made of tiny elements we call pixels. Since our computers only understand numbers, these pixels are given to the computer as a list of numbers.  
+
+To keep things simple, we will only deal with black and white images for now. So, for a black and white image, each pixel is a number between 0 and 1. 
+
+Try it out! Move the slider around to get a different shade of gray
+"""
+
+# ╔═╡ c4da04ab-f7e0-46fd-b352-e518e4733608
+g_slider = @bind g PlutoUI.Slider(0:0.05:1)
+
+# ╔═╡ 96a4b35a-5a3a-4ad0-9ffe-306db46d1c03
+#Move the Slider: $(g_slider) $(pixel)
+pixel = Gray.(g)
+
+# ╔═╡ eb983abe-5899-431b-84a1-7d06715d75c4
+md"""
+Now that you know how images are built. Let's get to the fun part. First let's start by getting a test image
+"""
+
+# ╔═╡ ef0bea69-6dfd-4526-a96e-6607771660fe
+testimage("coffee")
+
+# ╔═╡ 04bdc84a-0e3b-48d7-9129-50bc8c421e7c
+md"""Pretty neat right? But we're sticking with black and white images. """
+
+# ╔═╡ ebb63415-2738-47f3-b0ef-0fd6e6bac259
+md"""
+## Using Filter 
+
+Now to make things more interesting, we will add some code that applies a filter to this image."""
+
+# ╔═╡ fcf738bd-ab29-4ad8-a837-0fb2d0e7ed85
+test_img = Gray.(testimage("coffee"))
+
+# ╔═╡ 44d594ff-a25b-4455-be56-863345c67b68
+# Code from "Towards Data Science": https://towardsdatascience.com/understanding-convolution-by-implementing-in-julia-3ed744e2e933
+function conv(input, filter)
+    input_r, input_c = size(input)
+    filter_r, filter_c = size(filter)
+
+    result = zeros(input_r-filter_r+1, input_c-filter_c+1)
+    result_r, result_c = size(result)
+
+    for i in 1:result_r
+        for j in 1:result_c
+            result[i,j] = sum(input[i:i+filter_r-1,j:j+filter_c-1].*filter)
+        end
+    end
+  
+    return result
+end
+
+# ╔═╡ 065819eb-3697-417d-9251-d7537cd3d247
+md"""Can you guess what this filter does?"""
+
+# ╔═╡ f2d1dbd2-2d93-4f39-9204-5b321fbe9a67
+test_filter = [[1,1,1] [0,0,0] [-1,-1,-1]]
+
+# ╔═╡ 8ae0248f-6eba-4941-85f4-b21be3c7e725
+begin
+	height, width = size(test_img)
+	filtered_img = Gray.(conv(test_img, test_filter))
+	padded_img = test_img[2:height-1, 2:width-1]
+	[padded_img filtered_img]
+end
+
+# ╔═╡ a174850b-91cc-4463-ab28-61ca0a7221c6
+md"""Now it's your turn! Try changing the values of the filter matrix and see what happens!"""
+
+# ╔═╡ ed87ff94-64e2-4fdf-ab57-cd037a9b6eea
+md""" $(@bind a1 Scrubbable(-1:0.05:1)) | $(@bind a2 Scrubbable(-1:0.05:1)) | $(@bind a3 Scrubbable(-1:0.05:1))
+ 
+ $(@bind b1 Scrubbable(-1:0.05:1)) | $(@bind b2 Scrubbable(-1:0.05:1)) | $(@bind b3 Scrubbable(-1:0.05:1))
+
+ $(@bind c1 Scrubbable(-1:0.05:1)) | $(@bind c2 Scrubbable(-1:0.05:1)) | $(@bind c3 Scrubbable(-1:0.05:1))
+
+"""
+
+# ╔═╡ b26f85aa-596d-44bd-b356-afec3077693a
+filter = [[a1, a2, a3] [b1, b2, b3] [c1, c2, c3]]
```

Comment:
the @bind of the matrix is transposed to the filter you display here

- you could try `Scrubbable(Matrix)` as Fons proposed

- as default values, you could also think to use the zero matrix but have the middle [2,2,] entry be "1" by default.

- You could offer some tasks here e.g. "can you smooth the image?" "can you find a flilter to sharpen it?"


- you could also think of trying a game, e.g. "can you smooth the image" and make a box green if they succeed or something. But I dont know how easy this is a) in pluto and b) conceptually, if there are multiple ways to smooth